### DISCUSSION TITLE
Corpus compact

### DISCUSSION THREAD

**Discussion by albin02t** [JasonKessler/scattertext#112] [Q&A] [answered]
What does the st.CorpusFromPandas().compact(st.AssociationCompactor(2000)) do exactly? and how does it function?
Also what are the other functions such as st.ClassPercentageCompactor()
Couldn't find any explanation in the source code

**Comment by JasonKessler** [comment, marked as answer]
It accumulates the highest scoring words (most associated) for each category, as selected by a set frequency rank in all classes, until no more than 2000 (or whatever the specified number is) words are collected. 

By default, Scaled F-Score is used, but any `TermScorer` such as `RankDifference`, `DeltaJSDivergence`, etc. could be used,
